plugins/daapclient/__init__.py

```diff
@@ -302,24 +309,28 @@ def rebuild_share_menu_items(self):
             else:
                 service_name = info.name
 
-            # Retrieve IP address(es)
-            if False:
-                # Display one entry for IPv4 and one for IPv6
-                addresses = info.parsed_addresses(zeroconf.IPVersion.All)
+            if ZEROCONF_LEGACY:
+                # Legacy mode: returns only a single IPv4 address
+                addresses = [socket.inet_ntoa(info.address)]
             else:
-                # Display an entry for either IPv4 or IPv6, depending
-                # on the setting
-                if show_ipv6:
-                    addresses = info.parsed_addresses(zeroconf.IPVersion.V6Only)
+                # NOTE: in its current implementation (v.0.25.1), zeroconf
+                # appears to always return at most one IPv4 and one IPv6
+                # address, even if the service advertises multiple addresses.
+                # This appears to be tied to record caching, which keeps
+                # track of only the last parsed IPv4 and IPv6 address.
+
+                if False:
```

Comment:
I've simplified this block a bit as I realized that the Avahi backend shows IPv6 entries in addition to IPv4, not instead of them.

Ideally, we would get an entry per published IP address (although perhaps that would not be ideal in the case of IPv6), but as noted in the comment, python-zeroconf at the moment appears to be returning at most one address of each type.